UITests/UITests.Tests.Shared/UITestBase.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Windows.Apps.Test.Foundation.Controls;
+using Windows.ApplicationModel.AppService;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Common;
+using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Infra;
+using System.Diagnostics;
+
+#if USING_TAEF
+using WEX.Logging.Interop;
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+#endif
+
+namespace UITests.Tests
+{
+    public abstract class UITestBase
+    {
+        public static TestApplicationInfo WinUICsUWPSampleApp
+        {
+            get
+            {
+                string assemblyDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+                string baseDirectory = Path.Combine(Directory.GetParent(assemblyDir).Parent.Parent.Parent.Parent.FullName, "UITests.App");
+
+                Log.Comment($"Base Package Search Directory = \"{baseDirectory}\"");
+
+                var exclude = new[] { "Microsoft.NET.CoreRuntime", "Microsoft.VCLibs", "Microsoft.UI.Xaml", "Microsoft.NET.CoreFramework.Debug" };
+                var files = Directory.GetFiles(baseDirectory, "*.msix", SearchOption.AllDirectories).Where(f => !exclude.Any(Path.GetFileNameWithoutExtension(f).Contains));
+
+                if (files.Count() == 0)
+                {
+                    throw new Exception(string.Format("Failed to find '*.msix' in {0}'!", baseDirectory));
+                }
+
+                string mostRecentlyBuiltPackage = string.Empty;
+                DateTime timeMostRecentlyBuilt = DateTime.MinValue;
+
+                foreach (string file in files)
+                {
+                    DateTime fileWriteTime = File.GetLastWriteTime(file);
+
+                    if (fileWriteTime > timeMostRecentlyBuilt)
+                    {
+                        timeMostRecentlyBuilt = fileWriteTime;
+                        mostRecentlyBuiltPackage = file;
+                    }
+                }
+
+                return new TestApplicationInfo(
```

Comment:
To aid clarity, adding the parameter names here helps me understand what's being passed.
```cs
                return new TestApplicationInfo(
                    testAppPackageName: "UITests.App",
                    testAppName: "3568ebdf-5b6b-4ddd-bb17-462d614ba50f_gspb8g6x97k2t!App",
                    testAppPackageFamilyName: "3568ebdf-5b6b-4ddd-bb17-462d614ba50f_gspb8g6x97k2t",
                    testAppMainWindowTitle: "UITests.App",
                    processName: "UITests.App.exe",
                    installerName: mostRecentlyBuiltPackage.Replace(".msix", string.Empty),
                    certSerialNumber: "24d62f3b13b8b9514ead9c4de48cc30f7cc6151d",
                    baseAppxDir: baseDirectory);
```